Automattic/woocommerce-payments
# Make updating existing customer details during checkout async

Fixes #4490 

#### Changes proposed in this Pull Request
- Make updating existing customer details during checkout async for performance benefits.










#### Testing instructions



- Follow the normal checkout flow(both UPE and non UPE) for an existing customer.
- Ensure there is a scheduled action created as `wcpay_update_customer_with_order_data` in Scheduled Actions (Tools > Scheduled Actions)
- Run the action and ensure there are no errors, additionally check logs to ensure a POST `wcpay/customers/{{customer_id}}` call was made (check WP-Admin > WooCommerce > Status > Logs and there should be a log for WooCommerce Payments with today's date)
- Ensure any UPE and subscription flows are not affected by this change. 
  - Context for subscriptions: p1658831357623559-slack-C02BW3Z8SHK


-------------------


**Post merge**
